logstash-core/src/test/java/org/logstash/secret/store/SecretStoreFactoryTest.java

```diff
@@ -0,0 +1,194 @@
+package org.logstash.secret.store;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.logstash.secret.SecretIdentifier;
+import org.logstash.secret.store.backend.JavaKeyStore;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.logstash.secret.store.SecretStoreFactory.*;
+
+/**
+ * Unit tests for {@link SecretStoreFactory}
+ */
+public class SecretStoreFactoryTest {
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    //exact copy from https://stackoverflow.com/questions/318239/how-do-i-set-environment-variables-from-java
+    //thanks @pushy and @Edward Campbell !
+    @SuppressWarnings("unchecked")
+    private static void setEnv(Map<String, String> newenv) throws Exception {
+        try {
+            Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
+            Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
+            theEnvironmentField.setAccessible(true);
```

Comment:
Do you need to suppress accessibility warnings also? (Java 9?)

Reply:
Java 8 isn't complaining, and it would prefer to wait to make Java 9 specific changes until we support Java 9.  FWIW, It should only be a warning in on test compilation.